### ISSUE TITLE
MandatoryBracesIfStatements: false negative on else

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5322]
## Expected Behavior
I can't decide if this is a false positive (for `if`) or a false negative (for `else`). What I'm sure about is that it looks inconsistent.

## Observed Behavior
> Detekt - MandatoryBracesIfStatements: Multi-line if statement was found that does not have braces. These braces should be added to improve readability.
![image](https://user-images.githubusercontent.com/2906988/191108363-20d7c8df-7b61-42ad-b71d-83500728d98c.png)

## Steps to Reproduce
```kotlin
fun test(a: Boolean, b: Boolean, c: Boolean) {
	if (a) // flagged:
		if (b) {
			println("")
		} else {
			println("")
		}
	else // not flagged:
		if (c) {
			println("")
		} else {
			println("")
		}
}
```

## Context
I was trying to write a nice repro for another issue, when I noticed that this is happening.

## Investigation
I was suspecting that the control flow would not check the else, because there's a reporting in the `if`, but this code doesn't flag it either:
```kotlin
fun test(a: Boolean, b: Boolean, c: Boolean) {
	if (a) {
		if (b) {
			println("")
		} else {
			println("")
		}
	} else
		if (c) {
			println("")
		} else {
			println("")
		}
}
```

## Your Environment

* Version of detekt used: 1.21.0

**Comment by zsfVishnu** [comment]
Building on the observation above, i noticed that the flagging is not present on else blocks only. Even for the 'if else' block, detekt is flagging the **MandatoryBracesIfStatements** violation 

`fun test(a: Boolean, b: Boolean, c: Boolean) {
    if (a) // flagged:
        if (b)
            println("") //  flagged:
        else
            println("") //  flagged:

    else if(b)//  flagged:
        if (c)
            println("") //  flagged:
         else
            println("") //  flagged:

    else // not flagged:
        if (c)
            println("") //  flagged:
        else
            println("") //  flagged:

}`

**Comment by zsfVishnu** [comment]
I think the reason for this inconsistent behaviour is that the rule `MandatoryBracesIfStatements` is only checking for if nodes. Even for `else if` blocks, the `if` is triggering the flag. I couldn't find a rule for checking the `else` statements, so i think i'll create a PR with a rule to check else blocks

**Comment by zsfVishnu** [comment]
<img width="1684" alt="Screenshot 2022-09-20 at 4 33 21 PM" src="https://user-images.githubusercontent.com/34836841/191241694-49af8d8f-392f-4b30-9dfe-95aa5a242368.png">

Verified the above hypothesis by running the above scenario in the test file

**Comment by zsfVishnu** [comment]
@TWiStErRob  My assumption on this is that somehow the last `else` and the `if` is treated as a single `else if` , and the subsequent `else` as part of the outer block. (Kind of like flattening it out). Even though this is theoretically correct, i strongly feel that since this is a styling rule, we ought to throw the same warning/error as we'd have for the if block.

**Comment by BraisGabin** [comment]
It's a false negative. The else should be flagged.

**Comment by zsfVishnu-d11** [comment]
Cool. @BraisGabin i've started working on a fix for the same.

**Comment by zsfVishnu** [comment]
Cool. @BraisGabin i've started working on a fix for the same.

**Comment by BraisGabin** [comment]
@zsfVishnu did you have time to work on this?

**Comment by VitalyVPinchuk** [comment]
@BraisGabin 
I could fix this in https://github.com/detekt/detekt/issues/5133

**Comment by zsfVishnu** [comment]
@BraisGabin I did not get the time to finish it. @VitalyVPinchuk , is it fixed in your PR?

**Comment by VitalyVPinchuk** [comment]
@zsfVishnu  It will be
